compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposePanel.desktop.kt

```diff
@@ -171,38 +172,58 @@ class ComposePanel @ExperimentalComposeUiApi constructor(
     }
 
     override fun add(component: Component): Component {
-        if (bridge == null) {
-            return component
+        addToLayer(component, componentLayer)
+        if (!interopBlending) {
+            bridge?.addClipComponent(component)
         }
-        val clipComponent = ClipComponent(component)
-        clipMap[component] = clipComponent
-        bridge!!.clipComponents.add(clipComponent)
-        return super.add(component, Integer.valueOf(0))
+        return component
     }
 
     override fun remove(component: Component) {
-        bridge!!.clipComponents.remove(clipMap[component]!!)
-        clipMap.remove(component)
+        bridge?.removeClipComponent(component)
         super.remove(component)
     }
 
+    private fun addToLayer(component: Component, layer: Int) {
+        if (renderApi == GraphicsApi.METAL && bridge !is SwingComposeBridge) {
```

Comment:
```suggestion
        if (renderApi == GraphicsApi.METAL && !renderOnGraphics) {
```

The panel knows what it creates and `renderOnGraphics` defines that it will create `SwingComposeBridge` (see `createComposeBridge`)

Reply:
It's already changed it in next refactor, so if it's not a big deal, I'd avoid adding extra conflicts.

Reply:
and one more moment - in theory system property can be changed externally after the creation, so such check is more robust.